lms/djangoapps/instructor_task/tests/test_tasks_helper.py

```diff
@@ -1711,7 +1719,14 @@ def test_certificate_generation_whitelist_already_generated(self):
         }
         self.assertCertificatesGenerated(task_input, expected_results)
 
-    def test_certificate_generation_whitelisted_not_generated(self):
+    @ddt.data(
+        (CertificateStatuses.downloadable, 3),
+        (CertificateStatuses.generating, 3),
+        (CertificateStatuses.notpassing, 5),
```

Comment:
Yeah, it's weird. My understanding is that the user only sees a cert if it's "downloadable". If you've just passed the course in audit mode, then you're not going to get a certificate. The point of the whitelist is to get people from whatever state they're in to "downloadable". So even though certificates "were generated" for 2 students with `notpassing` or `audit_passing`, those students didn't actually get certificates. They just got rows in the cert table